src/common/log.c

```diff
@@ -23,21 +23,21 @@
 #include <stdbool.h>
 
 #include "log.h"
-bool g_nolog = false;
-enum log_levels g_log_level = LOG_WARNING;
 
-static const char *log_level_name[] = { "INFO", "DEBUG", "WARN", "ERROR" };
+#ifndef LOG_LEVEL
+#define LOG_LEVEL LOG_WARNING
+#endif
+
+static const char *log_level_name[] = { "DEBUG", "INFO", "WARN", "ERROR" };
 
 void log_message(int l, const char *file, int line, const char *fmt, ...)
 {
 	va_list arg;
-	if (g_nolog) return;
-	if(g_log_level > l) return;
+	if(LOG_LEVEL > l) return;
```

Comment:
For the No log case.
if #define LOG_LEVEL LOG_NEVER
if log_message(LOG_NEVER,....)
look still log the message.

Reply:
You're not supposed to use LOG_NEVER in your "log_message" function.
If you don't want logging, just set LOG_LEVEL to LOG_NEVER then all valid log levels, from LOG_DEBUG to LOG_ERROR, will be ignored.

Reply:
yes, make sense